src/mscorlib/src/System/Number.cs

```diff
@@ -685,6 +685,14 @@ internal unsafe struct NumberBuffer {
 
         [System.Security.SecuritySafeCritical]  // auto-generated
         internal unsafe static Int32 ParseInt32(String s, NumberStyles style, NumberFormatInfo info) {
+            if (style == NumberStyles.Integer && info == NumberFormatInfo.InvariantInfo) {
```

Comment:
I did a couple of tests on desktop CLR (so that might be the reason for these results), but this check seems overly conservative to me:
```csharp
Thread.CurrentThread.CurrentCulture = CultureInfo.InvariantCulture;
// Very common path; reached with int.Parse
bool currentCulture = NumberFormatInfo.CurrentInfo == NumberFormatInfo.InvariantInfo;
// Less common path, but probably still common: reached with int.Parse with CultureInfo.InvariantCulture
bool invariantCulture = NumberFormatInfo.GetInstance(CultureInfo.InvariantCulture) ==
                        NumberFormatInfo.InvariantInfo;
```
Both of these give false on desktop CLR (even if it's different in CoreCLR, this makes it a lot harder to port this performance improvement).
I can't even find out how to hit this check in a way that it will pass. Am I missing something?

In any case even if the second case (with `CultureInfo.InvariantCulture`) would be true, I still think the check can be improved so it works for the common case `int.Parse` without a culture. For example, consider my own culture, `da-DK`: `NumberFormatInfo.NumberDecimalSeparator` for Danish is `,`, so this would never match the invariant info. However this information is not used at all for regular integer parsing, and my culture would still benefit from this new fast-path.